test/unit/util/mapbox.test.js

```diff
@@ -272,5 +273,137 @@ test("mapbox", (t) => {
+        });
+
+        t.test('does not POST when mapboxgl.ACCESS_TOKEN is not set', (t) => {
+            config.ACCESS_TOKEN = null;
+
+            mapbox.postTurnstileEvent([' a.tiles.mapxbox.com']);
+
+            t.equal(window.server.requests.length, 0);
+            t.end();
+        });
+
+        t.test('does not POST when urls does not point to mapbox.com', (t) => {
+            config.ACCESS_TOKEN = 'pk.*';
+
+            mapbox.postTurnstileEvent(['a.tiles.boxmap.com']);
+
+            t.equal(window.server.requests.length, 0);
+            t.end();
+        });
+
+        t.test('Does not POST appuserTurnstile event second time within same calendar day', (t) => {
+            //Depend on having a successful POST in a prior test.
+            config.ACCESS_TOKEN = 'pk.*';
+
+            mapbox.postTurnstileEvent(['a.tiles.mapbox.com']);
+
+            const req = window.server.requests[0];
+            t.false(req);
+
+            t.end();
+        });
+
+        t.test('POSTs appuserTurnstile event when access token changes', (t) => {
+            config.ACCESS_TOKEN = 'pk.new.*';
+
+            mapbox.postTurnstileEvent(['a.tiles.mapbox.com']);
+
+            const req = window.server.requests[0];
+            req.respond(200);
+
+            const reqBody = JSON.parse(req.requestBody)[0];
+            t.equal(req.url, `${config.EVENTS_URL}?access_token=pk.new.*`);
+            t.equal(req.method, 'POST');
+            t.equal(reqBody.event, 'appUserTurnstile');
+            t.equal(reqBody.sdkVersion, version);
+            t.ok(reqBody.userId);
+
+            t.end();
+        });
+
+        t.test('POSTs appuserTurnstile event on next calendar day', (t) => {
+            config.ACCESS_TOKEN = 'pk.*';
+            const today = Date.now();
+            const tomorrow = today + (25 * 60 * 60 * 1000); // Add a day
+            t.stub(browser, 'now').callsFake(() => tomorrow);
+
+            mapbox.postTurnstileEvent(['a.tiles.mapbox.com']);
+
+            const req = window.server.requests[0];
+            req.respond(200);
+
+            const reqBody = JSON.parse(req.requestBody)[0];
+            t.equal(req.url, `${config.EVENTS_URL}?access_token=pk.*`);
+            t.equal(req.method, 'POST');
+            t.equal(reqBody.event, 'appUserTurnstile');
+            t.equal(reqBody.sdkVersion, version);
+            t.ok(reqBody.userId);
+            t.ok(reqBody.created, new Date(tomorrow).toISOString());
+
+            t.end();
+        });
+
+        t.test('POSTs consecutive appuserTurnstile events on different calendar days', (t) => {
+            config.ACCESS_TOKEN = 'pk.*';
+            const today = Date.now();
+            const laterToday = today + 1;
+            const tomorrow = today + (25 * 60 * 60 * 1000); // Add a day
+
+            const stub = t.stub(browser, 'now');
+            stub.onCall(0).returns(today);
+            stub.onCall(1).returns(laterToday);
+            stub.onCall(2).returns(tomorrow);
```

Comment:
This approach to stubbing could be fragile, because it assumes that `browser.now` is only called once per `postTurnstileEvent` call.  Alternative:

```js
let now = Date.now();

t.stub(browser, 'now').callsFake(() => now);

mapbox.postTurnstileEvent(['a.tiles.mapbox.com']);
const reqToday = window.server.requests[0];
reqToday.respond(200);
let reqBody = JSON.parse(reqToday.requestBody)[0];
t.ok(reqBody.created, new Date(now).toISOString());

now += 1; // later today
mapbox.postTurnstileEvent(['b.tiles.mapbox.com']);
t.ok(window.server.requests.length === 1);

now += (25 * 60 * 60 * 1000); // tomorrow
mapbox.postTurnstileEvent(['c.tiles.mapbox.com']);
const reqTomorrow = window.server.requests[1];
reqTomorrow.respond(200);
reqBody = JSON.parse(reqTomorrow.requestBody)[0];
t.ok(reqBody.created, new Date(now).toISOString());
```

Reply:
The point of this test is to test the queuing mechanism(as suggested by @jfirebaugh [previously(comment)](https://github.com/mapbox/mapbox-gl-js/pull/6980/files/ab94794b5a354d4805f6d1875c3948fb78082636#r204520204). Events are queued correctly triggered consecutively and/or responses are returned with a higher latency. The alternative you suggest is a serial approach that is being tested already.

Reply:
Ah, I see.  We should be able to use the more explicit approach to manipulating mock-time though, right?